lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControlModule.java

```diff
@@ -14,44 +14,24 @@
 package io.trino.plugin.base.security;
 
 import com.google.inject.Binder;
-import com.google.inject.Inject;
-import com.google.inject.Module;
-import com.google.inject.Provides;
-import io.airlift.log.Logger;
-import io.trino.plugin.base.CatalogName;
-import io.trino.spi.connector.ConnectorAccessControl;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
 
-import java.io.File;
-
-import static com.google.common.base.Suppliers.memoizeWithExpiration;
-import static io.airlift.configuration.ConfigBinder.configBinder;
-import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
 
 public class FileBasedAccessControlModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
-    private static final Logger log = Logger.get(FileBasedAccessControlModule.class);
-
     @Override
-    public void configure(Binder binder)
+    public void setup(Binder binder)
     {
-        configBinder(binder).bindConfig(FileBasedAccessControlConfig.class);
-    }
+        install(conditionalModule(
+                BaseAccessControlConfig.class,
+                config -> config.isRest(),
```

Comment:
Instead of using `config#isRest` can we introduce a `enum` like `FILE` or `REST` - and install the modules based on the type ?

Reply:
SecuritySource enum introduced